### ISSUE TITLE
datadog deriving tags

### ISSUE THREAD

**Issue by InoMurko** [ivx/ink#50]
Hey! Wonderful work!

We've been trying to adapt this backend logger in our project as well at OmiseGO. We're using Datadog as the go-to metric, logging backend platform.

One issue we're seeing is that Ink sends `hostname` as part of it's `base` log. Now this confuses datadog because `hostname` is one of their reserved words. What does that mean? It means that all tags that their agent would normally parse from the infrastructure and environment (like cluster location, kubernetes pods, etc.) now get lost.

I've quickly patched by removing hostname from the base_map and all tags magically started appearing. I wonder what your use case for hostname is and what would be the appropriate approach to mitigate this?

Cheers
Ino

**Comment by mmainz** [comment]
We don't use the `hostname` for anything specific, but we decided to have the same log format for all our services. And that format is [node-bunyan](https://github.com/trentm/node-bunyan), which has `hostname` by default.

That is also the reason why we log the PID of processes as `erlang_pid`, because `pid` is a default field in `node-bunyan`.

I have no experience with datadog, so I assume you can't fix this on their side. Maybe you could make the base log configurable and keep the current base log as the default? I'd be open to accept a PR that implements this.

**Comment by InoMurko** [comment]
I think we can solve this pickle pretty quickly, happy to make a PR.
Any preference on the design? Would the following work for you?
- provide a config `exclude_hostname: true` where the default is `false`
- a test:
```
 test "respects exclude hostname " do
    Logger.configure_backend(Ink, exclude_hostname: :true)
    Logger.info("test")

    assert_receive {:io_request, _, _, {:put_chars, :unicode, msg}}
    assert msg |> Jason.decode!() |> Map.get("hostname", :excluded) == :excluded
  end
```
- expanding `base_map/4` to `base_map/5` accepting `config` as the last argument

Well... I'll let you review the PR:
https://github.com/ivx/ink/pull/51

**Comment by mmainz** [comment]
@InoMurko I just published 1.0.3 which includes your PR. Let me know if it works.
